src/main/java/com/brightpath/learnify/domain/comment/CommentService.java

```diff
@@ -0,0 +1,53 @@
+package com.brightpath.learnify.domain.comment;
+
+import com.brightpath.learnify.domain.auth.PermissionAccessService;
+import com.brightpath.learnify.domain.common.ResourceType;
+import com.brightpath.learnify.domain.common.UuidProvider;
+import com.brightpath.learnify.domain.quiz.comment.Comment;
+import com.brightpath.learnify.domain.quiz.comment.CommentCreation;
+import com.brightpath.learnify.exception.authorization.UserNotAuthorizedToAddCommentException;
+import com.brightpath.learnify.exception.authorization.UserNotAuthorizedToViewCommentsException;
+import com.brightpath.learnify.persistance.comment.CommentEntity;
+import com.brightpath.learnify.persistance.comment.CommentRepository;
+import com.brightpath.learnify.persistance.common.PersistentMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+import static com.brightpath.learnify.domain.auth.permission.ResourceAccessEnum.READ_ONLY;
+
+@Service
+@RequiredArgsConstructor
+public class CommentService {
+    private final PersistentMapper persistentMapper;
+    private final UuidProvider uuidProvider;
+    private final CommentRepository commentRepository;
+    private final PermissionAccessService permissionAccessService;
+
+    public List<Comment> getCommentsForResource(UUID resourceId, ResourceType resourceType) {
+        if (!permissionAccessService.checkUserPermissionToViewResource(resourceId, resourceType)) {
```

Comment:
Why not at the REST controller level with `PreAuthorize` annotation?
And as you wrote, there is no special permissions to view comments so why additional exceptions?
We can use permission check for generic resource as for permissions controller:
```
@PreAuthorize("""
                    @permissionAccessService.checkUserPermissionToViewResource(#resourceId, #resourceType.name()) or
                    @userIdentityService.isCurrentUserAdmin()
            """)
```

Reply:
nice idea, I didn't know we can use it when we do not know the exact resource